### PR TITLE
[Ray State API] Include `ray_namespace` in `ActorState`

### LINKED ISSUE #28544: [Ray State API] Include `namespace` in `ActorState`
### Description

**Background**

`ActorState` provides useful information about actors, such as `name` and `actor_id`.

**Request**

`ActorState` should include `namespace`.

### Use case

The [Serve controller relies on `list_named_actors`](https://github.com/ray-project/ray/blob/c5952f2163ea24fa072b1e57e4fa37b5caf1a56c/python/ray/serve/controller.py#L124-L131) to detect which deployment replicas are currently running. It uses the namespaces to distinguish between Serve actors (such as HTTPProxies and deployment replicas) and any other Ray actors that might be running in the cluster.

To use the corresponding `list_actors` call from the new State API, the controller would need to know which actors belong to which namespaces.

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

This change adds `ray_namespace` to the `ActorState` dataclass. This allows functions like `list_actors` to return actor namespaces.

## Related issue number

Closes #28544.

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
     - A new unit test is added to `test_state_api.py`.

### CODE REVIEW

File: python/ray/tests/test_state_api.py [hunk 1]
```diff
@@ -1600,6 +1601,29 @@ def verify():
     print(list_actors())
 
 
+def test_list_actors_namespace(shutdown_only):
```
Comment (rkooo567) [inline python/ray/tests/test_state_api.py:1604]: Consider adding this check to existing list actor test? (to reduce ray.init load on tests)

**Review by rkooo567 — APPROVED** [review]

File: python/ray/tests/test_state_api.py (hunk 1 above)
Comment (shrekris-anyscale) [reply python/ray/tests/test_state_api.py]: Good idea! I'm worried about cluttering the existing test with extra checks for a single field. I refactored the two unit tests to share a single class-scope fixture, so they both share a single `ray.init()` call. How does it look?

File: python/ray/tests/test_state_api.py [hunk 2]
```diff
@@ -1570,34 +1571,53 @@ def verify_output(cmd, args: List[str], necessary_substrings: List[str]):
     sys.platform == "win32",
     reason="Failed on Windows",
 )
-def test_list_get_actors(shutdown_only):
-    ray.init()
+class TestListActors:
```
Comment (rickyyx) [inline python/ray/tests/test_state_api.py:1574]: Ah nice - guess we should do this for some other tests as well.

**Review by rickyyx — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/jerology/">Jerology</a> on <a href="https://giphy.com/gifs/jerology-awesome-sam-didier-frickin-xTiTnBMEz7zAKs57LG">GIPHY</a></div>

**Pull request merged**